test/nodes/test_multi_node_weighted_sampler.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import itertools
+from enum import unique
+
+from parameterized import parameterized
+from torch.testing._internal.common_utils import TestCase
+from torchdata.nodes.adapters import IterableWrapper
+
+from torchdata.nodes.samplers.multi_node_weighted_sampler import MultiNodeWeightedSampler
+from torchdata.nodes.samplers.utils import StopCriteria
+
+from .utils import DummyIterableDataset, run_test_save_load_state
+
+
+class TestMultiNodeWeightedSampler(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self._num_samples = 10
+        self._num_datasets = 4
+        self._weights_fn = lambda i: 0.1 * (i + 1)
+        self._num_epochs = 3
+
+        self.datasets = {
+            f"ds{i}": IterableWrapper(DummyIterableDataset(self._num_samples, f"ds{i}"))
+            for i in range(self._num_datasets)
+        }
+        self.weights = {f"ds{i}": self._weights_fn(i) for i in range(self._num_datasets)}
+
+    def _setUpMultiNodeWeightedSampler(
```

Comment:
nit: Use underscores when python (except for setUp which we don't have a choice in).
```suggestion
    def _setup_multi_node_wighted_sampler(
```